Fixes #37560 - rewrite React GH action

jeremylenz opened the pull request:
#### What are the changes introduced in this pull request?

Change the 'React Tests' GitHub action to use @theforeman/actions in line with Foreman CI.

This will also allow us to use `foreman_version` to test Katello PRs against any Foreman PR or branch.

#### Considerations taken when implementing this change?

Also removed '(Non-blocking)' from the name, so it's assumed this is a blocking check. But I think we need a GH org admin to make that actually true.

#### What are the testing steps for this pull request?

critique my beautiful YAML
verify that CI passes and that the output reflects actual test runs

ekohl commented on .github/matrix.json:
Please don't include this. The whole design we've applied is that we take the Foreman core repository to define these. That way you can simply point to Foreman x.y and get a test matrix.

jeremylenz replied:
Removed (was a leftover from a previous iteration).

ekohl commented on .github/workflows/react_tests.yml:
```diff
@@ -1,45 +1,23 @@
+    uses: theforeman/actions/.github/workflows/foreman_plugin_js.yml@v0
+    with:
+      plugin: katello
+      foreman_version: develop # set to the Foreman release branch after branching :)
```
This is good, but we should also include it by default in `.github/workflows/ruby.yml` so we can simply update it on branching (instead of relying on adding it in branching).

jeremylenz commented:
```
build Expected — Waiting for status to be reported
```

I'm not sure what's wrong here.

ekohl commented:
> I'm not sure what's wrong here.

There's probably a required check configured for this branch that now no longer exists. Some admin needs to fix that.

evgeni commented on .github/workflows/react_tests.yml:
```diff
@@ -1,45 +1,23 @@
-name: React Tests (non-blocking)
```
This, albeit the name, used to be a required check in GH.
After the rename, it's not matching anymore, so you get the "waiting" job.

I don't think this *needs* to be a "required check", but if you want it to be, it needs to run on every PR, not limited to `paths` like below.

jeremylenz replied:
Ahhh, I thought it was my rename of `build` to `react-tests`. That explains why changing it back to `build` didn't work.

@parthaa I would like to call this "React Tests" (without the "(non-blocking)"). Can you please make "React Tests" the required check and remove "React Tests (non-blocking)" ?

ekohl replied:
@jeremylenz you probably need to merge this first before you can add it as a required check.

jeremylenz commented:
I had Partha remove the `build` required check and add a new required check for `react-tests`. I renamed the job here to `react-tests` and also removed the `paths` so it now runs on all pull requests. Let's see how it goes..

jeremylenz commented:
Looks like we might still have to merge this before the expected check is satisfied..

ekohl commented:
> Looks like we might still have to merge this before the expected check is satisfied..

That's also my experience
